scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -279,3 +280,57 @@ def get_generated_keys(self) -> List[str]:
         for operand in self._operands:
             generated_keys += operand.get_generated_keys()
         return generated_keys
+
+
+class AttackPointIteratorCartesianProduct(AttackPointIterator):
+    """
+    Attack point iterator cartesian product class. This class takes any amount
+    of operands and combines them just like a cartesian product would.
+    """
+
+    def __init__(self, operation: str, operands: List[Dict[str, Any]]) -> None:
+        """Initialize the cartesian product iterator.
+          
+          Args:
+            operation (str): The operation of the iterator
+                represents what the iterator does and what 
+                has to be in the config file. This is only used once to
+                double check if the operation is the correct one.
+                
+            operands (List[Dict[str, Any]]): The operands are any number of
+                iterator configs that will be combined."""
+        assert "cartesian_product" == operation
+        self._operands = list(
+            build_attack_points_iterator(operand) for operand in operands)
+
+        operand_lengths = [operand._len for operand in self._operands]
+        if not self._operands or any(
+                length <= 0
+                for length in operand_lengths) or len(operand_lengths) != 2:
+            self._len = 0
+            print(len(self))
+        else:
+            self._len = operand_lengths[0] * operand_lengths[1]
+
+    def __iter__(self) -> AttackPointIteratorT:
+        if any(len(operand) <= 0 for operand in self._operands) or len(
+                self._operands) != 2:
+            raise ValueError
+        return iter(
+            self.merge_dictionaries(value_one, value_two)
+            for value_one in self._operands[0]
+            for value_two in self._operands[1])
+
+    @staticmethod
+    def merge_dictionaries(value_one: Dict[str, Any],
```

Comment:
Why is this method being copy-pasted? It can easily go to the parent class. Also please rename it to `_merge_dictionaries` since the user is not expected to call it.

Reply:
But it isn't used by every child class is it ok to put it in the parent then?

Reply:
Fine by me, definitelly less wrong than duplicating code. You can also make it a free function if you wish.